metricbeat/module/zookeeper/connection/data.go

```diff
@@ -0,0 +1,108 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package connection
+
+import (
+	"bufio"
+	"io"
+	"regexp"
+	"strconv"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/metricbeat/mb"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+var capturer = regexp.MustCompile(`/(?P<ip>.*):(?P<port>\d+)\[(?P<interest_ops>\d*)]\(queued=(?P<queued>\d*),recved=(?P<received>\d*),sent=(?P<sent>\d*)\)`)
+
+func (m *MetricSet) parseCons(i io.Reader) ([]mb.Event, error) {
+	scanner := bufio.NewScanner(i)
+
+	result := make([]mb.Event, 0)
+
+	for scanner.Scan() {
+		metricsetFields := common.MapStr{}
+		rootFields := common.MapStr{}
+		line := scanner.Text()
+
+		oneParsingIsCorrect := false
+		keyMap, err := lineToMap(line)
+		if err != nil {
+			m.Logger().Debugf(err.Error())
+			continue
+		}
+
+		for k, v := range keyMap {
+			if k == "ip" {
+				if _, err := rootFields.Put("client.ip", v); err != nil {
+					m.Logger().Debugf("%v. Error placing key 'ip' on event", err)
+				} else {
+					oneParsingIsCorrect = true
+				}
+			} else if k == "port" {
+				m.checkRegexAndSetInt(rootFields, v, "client.port", &oneParsingIsCorrect)
+			} else {
+				m.checkRegexAndSetInt(metricsetFields, v, k, &oneParsingIsCorrect)
+			}
+		}
```

Comment:
Could `schema` be used here for these mappings?

Reply:
Lets leave this change for a follow up as this is blocking other things.

Reply:
Awesome, thanks Jaime!